### PR TITLE
chore: 🤖 refactored onboarding actions

### PR DESCRIPTION
**Pull request by hamzabell**

### CODE REVIEW

File: app/features/onboarding/onboarding-actions.server.ts [hunk 1]
```diff
@@ -17,35 +15,33 @@ import {
...
+import { requireUserNeedsOnboarding } from './onboarding-helpers.server';
+
+export const onboardingUserProfileAction = async ({
+  request,
+}: Pick<ActionFunctionArgs, 'request'>) => {
+  const user = await requireUserNeedsOnboarding(request);
+  const data = await validateFormDataFromRequest({
+    request,
+    schema: onboardingUserProfileSchema,
+  });
```
Comment (janhesters) [inline app/features/onboarding/onboarding-actions.server.ts:27]: Can these be parallelized?

File: app/features/onboarding/onboarding-actions.server.ts [hunk 2]
```diff
@@ -17,35 +15,33 @@ import {
   onboardingOrganizationSchema,
   onboardingUserProfileSchema,
 } from './onboarding-client-schemas';
-import { withUserRequiringOnboarding } from './onboarding-middleware.server';
+import { requireUserNeedsOnboarding } from './onboarding-helpers.server';
+
+export const onboardingUserProfileAction = async ({
+  request,
+}: Pick<ActionFunctionArgs, 'request'>) => {
+  const user = await requireUserNeedsOnboarding(request);
```
Comment (janhesters) [inline app/features/onboarding/onboarding-actions.server.ts:23]: Also, is `requireUserNeedsOnboarding` efficient aka. parallel? We probably want to make sure it is and if NOT refactor it to be that in a different PR.

File: app/utils/parse-form-data.server.ts [hunk 3]
```diff
@@ -97,3 +97,21 @@ export const withValidatedFormData =
...
+ * @throws Will throw a 400 Bad Request error if the form data does not conform
+ * to the schema with an  object containing the formatted error details.
+ */
+export const validateFormDataFromRequest = async <Schema extends ZodType>({
+  request,
+  schema,
+}: {
+  request: Request;
+  schema: Schema;
+}) => await parseFormData(schema, request);
```
Comment (janhesters) [inline app/utils/parse-form-data.server.ts:117]: This just seems like an object wrapper around `parseFormData`. Should we refactor `parseFormData` to be this object wrapper instead?

**Review by janhesters — APPROVED** [review]
Approved, but with some comments.

File: app/utils/parse-form-data.server.ts (hunk 3 above)
Comment (janhesters) [reply app/utils/parse-form-data.server.ts:117]: This comment seems unaddressed 🤔

Comment (hamzabell) [reply app/utils/parse-form-data.server.ts:117]: I missed this comment, I would redact this feature

**Review by janhesters — APPROVED** [review]

File: playwright/e2e/organizations/organizations-settings-team-members.spec.ts [hunk 4]
```diff
@@ -335,11 +335,14 @@ test.describe('organizations settings team members page', () => {
...
         await memberListItem.getByRole('button', { name: /member/i }).click();
         const role = roleIterator.next().value!;
+
         await page
           .getByRole('option', { name: role })
           .getByRole('button')
           .click();
-        await page.keyboard.press('Escape');
+        await page.keyboard.down('Escape');
+        await new Promise(resolve => setTimeout(resolve, 1000));
```
Comment (janhesters) [inline playwright/e2e/organizations/organizations-settings-team-members.spec.ts:344]: Should we use `waitForTimeout` instead? https://playwright.dev/docs/next/api/class-page#page-wait-for-timeout

**Review by janhesters — APPROVED** [review]

**Review by janhesters — APPROVED** [review]
